### ISSUE TITLE
Docker service crash

### ISSUE THREAD

**Issue by mrmm** [coreos/coreos-kubernetes#770]
Hi I am not sure this is the right repo to put my question in

I am running Kubernetes cluster using CoreOS on top of AWS

And sometimes docker service on some nodes crash and every Pod that got scheduled on the node stay in Pending status.
So to solve the problem I terminate the node so the ASG recreate a new one, this problem  happened about 5 times the last month in the cluster.

I dont know what is the problem, i have tried to restart the docker service but `systemctl` stay blocked.

```
Here is the `systemctl status docker` after executing `systemctl restart docker`
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib64/systemd/system/docker.service; disabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/docker.service.d
           └─40-flannel.conf
   Active: deactivating (stop-sigkill) (Result: timeout) since Wed 2016-12-07 13:46:27 UTC; 2min 12s ago
     Docs: http://docs.docker.com
 Main PID: 1046 (docker)
    Tasks: 2
   Memory: 1.6G
      CPU: 1h 37min 2.665s
   CGroup: /system.slice/docker.service
           ‣ 1046 [docker]

Dec 06 17:55:34 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-06T17:55:34.849543947Z" level=info msg="Container 766f2401692d7b14bcb9c5df7f5e8112ce1c72d233086756ed79598917c9eb45 failed to exit within 30 seconds of S
Dec 07 09:03:31 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T09:03:31.556026033Z" level=error msg="Handler for GET /images/xxxxx.dkr.ecr.eu-west-1.amazonaws.com/xx-xx-xx-crm-server-master-81
Dec 07 13:12:06 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:12:06.864987491Z" level=error msg="Handler for GET /images/xxxx.dkr.ecr.eu-west-1.amazonaws.com/xx-xx-xx-crm-server-develop-7
Dec 07 13:46:27 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:27.166963431Z" level=info msg="Processing signal 'terminated'"
Dec 07 13:46:27 ip-172-16-220-65.eu-west-1.compute.internal systemd[1]: Stopping Docker Application Container Engine...
Dec 07 13:46:37 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:37.167414272Z" level=info msg="Container cc5d266dbb134c74aa62bc09090d336165f9af6a5bc2be628b93cdb0d774a005 failed to exit within 10 seconds of S
Dec 07 13:46:37 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:37.178950383Z" level=info msg="Container 74e3f883a28edc171a4efa3151f44559b0ba3808fc0247367fa8d49d82376222 failed to exit within 10 seconds of S
Dec 07 13:46:37 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:37.179077150Z" level=info msg="Container 6eace5f782b31d2b20389756081efa44056051f9f07f2c1f52ebb58e73627292 failed to exit within 10 seconds of S
Dec 07 13:46:42 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:42.167204320Z" level=error msg="Force shutdown daemon"
Dec 07 13:47:57 ip-172-16-220-65.eu-west-1.compute.internal systemd[1]: docker.service: State 'stop-sigterm' timed out. Killing.
lines 1-23/23 (END)Connection to 52.31.90.61 closed by remote host.
```

And this is the journalctl -u docker.service

```
aws.com/play-retail-server:v2
Dec 07 13:46:27 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:27.166963431Z" level=info msg="Processing signal 'terminated'"
Dec 07 13:46:27 ip-172-16-220-65.eu-west-1.compute.internal systemd[1]: Stopping Docker Application Container Engine...
Dec 07 13:46:37 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:37.167414272Z" level=info msg="Container cc5d266dbb134c74aa62bc09090d336165f9af6a5bc2be628b93cdb0d774a005 failed to exit within 10 seconds of SIGTERM - using the force"
Dec 07 13:46:37 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:37.178950383Z" level=info msg="Container 74e3f883a28edc171a4efa3151f44559b0ba3808fc0247367fa8d49d82376222 failed to exit within 10 seconds of SIGTERM - using the force"
Dec 07 13:46:37 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:37.179077150Z" level=info msg="Container 6eace5f782b31d2b20389756081efa44056051f9f07f2c1f52ebb58e73627292 failed to exit within 10 seconds of SIGTERM - using the force"
Dec 07 13:46:42 ip-172-16-220-65.eu-west-1.compute.internal dockerd[1046]: time="2016-12-07T13:46:42.167204320Z" level=error msg="Force shutdown daemon"
```

I could not debug the source problem.

**Comment by pieterlange** [comment]
Hi,

kube-aws moved to https://github.com/coreos/kube-aws

As for your problem: this happens sometimes with docker. What version are you running?
https://github.com/kubernetes/node-problem-detector has been made to fix these kind of problems.

**Comment by mrmm** [comment]
@pieterlange The versions are :

 - docker version
```
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.4
 Git commit:   1f8f545
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.4
 Git commit:   1f8f545
 Built:
 OS/Arch:      linux/amd64
```
 - kubectl version :
```
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.4", GitCommit:"dd6b458ef8dbf24aff55795baa68f83383c9b3a9", GitTreeState:"clean", BuildDate:"2016-08-01T16:45:16Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.4+coreos.0", GitCommit:"be9bf3e842a90537e48361aded2872e389e902e7", GitTreeState:"clean", BuildDate:"2016-08-02T00:54:53Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by pieterlange** [comment]
The docker version included in coreos stable is 1.11.2 so you must be running an old AMI.

There's a lot of improvements in kube-aws at the new repo, including some features to make it easier to upgrade as well.

**Comment by mrmm** [comment]
Thank you very, much I will try to use the latest version ;)
